app/src/main/java/com/humberto/tasky/agenda/data/helper/WorkerHelper.kt

```diff
@@ -0,0 +1,34 @@
+package com.humberto.tasky.agenda.data.helper
+
+import androidx.work.Constraints
+import androidx.work.ListenableWorker
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.workDataOf
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class WorkerHelper @Inject constructor(
+    val workManager: WorkManager
+) {
+
+    inline fun <reified T: ListenableWorker> enqueueSyncPendingWorker(
+        idKey: String,
+        id: String
+    ) {
+        val request = OneTimeWorkRequestBuilder<T>()
+            .setInputData(
+                workDataOf(idKey to id)
+            )
+            .setInitialDelay(15, TimeUnit.MINUTES)
+            .setConstraints(
+                Constraints(
+                    requiredNetworkType = NetworkType.CONNECTED
+                )
+            )
+            .build()
+
+        workManager.enqueue(request)
+    }
```

Comment:
How about adding a retry policy to retry a worker at a later time to better ensure that the work gets executed?

```
  val workRequest = OneTimeWorkRequestBuilder<T>()
    .addTag(FetchFullAgendaWorker.TAG)
    .setConstraints(
      Constraints.Builder()
        .setRequiredNetworkType(NetworkType.CONNECTED)
        .build()
    )
    .setBackoffCriteria(
      backoffPolicy = BackoffPolicy.EXPONENTIAL,
      backoffDelay = 2000L,
      timeUnit = TimeUnit.MILLISECONDS
    )
    .build()
```